src/pybind/mgr/cephadm/templates/services/mgmt-gateway/external_server.conf.j2

```diff
@@ -0,0 +1,76 @@
+
+server {
+{% if spec.disable_https %}
+    listen {{ spec.port or 80 }};
+{% else %}
+    listen                    {{ spec.port or 443 }} ssl;
+    listen                    [::]:{{ spec.port or 443 }} ssl;
+    ssl_certificate            /etc/nginx/ssl/nginx.crt;
+    ssl_certificate_key /etc/nginx/ssl/nginx.key;
+    {% if spec.ssl_protocols %}
+    ssl_protocols            {{ spec.ssl_protocols | join(' ') }};
+    {% else %}
+    ssl_protocols            TLSv1.2 TLSv1.3;
```

Comment:
By default I think we should drop 1.2 (that's what we did in the Dashboard). Some cyphers in 1.2 are considered insecure, so you should trigger an health check alert if someone uses a vulnerable cypher.

That said, in some cases, people explicitly require 1.2 support (e.g.: they use some HW load balancers/VPNs or any network devices that don't have 1.3 support).

Reply:
I removed TLSv1.2 from the default .. anyway if some user wants to use it they can use a custom spec file.